super-csv/src/test/java/org/supercsv/io/TokenizerTest.java

```diff
@@ -586,6 +587,178 @@ public void testQuotedSpaces() throws Exception {
+				.build();
+
+		final String input = "\"field with \\\\ escape char\"";
+		tokenizer = createTokenizer(input, csvPref);
+		tokenizer.readColumns(columns);
+
+		assertTrue(columns.size() == 1);
+		assertEquals("field with \\ escape char", columns.get(0));
+		assertEquals(input, tokenizer.getUntokenizedRow());
+	}
+
+	/**
+	 * Tests the readColumns() method with an even number of escape characters
+	 * char.
+	 */
+	@Test
+	public void testEvenSeriesOfEscapeChars() throws Exception {
+
+		final CsvPreference csvPref = new CsvPreference.Builder(NORMAL_PREFERENCE)
+				.setQuoteEscapeChar('\\')
+				.build();
+
+		final String input = "\"\\\\\\\\\\\\\"";
+		tokenizer = createTokenizer(input, csvPref);
+		tokenizer.readColumns(columns);
+
+		assertTrue(columns.size() == 1);
+		assertEquals("\\\\\\", columns.get(0));
+		assertEquals(input, tokenizer.getUntokenizedRow());
+	}
+
+	/**
+	 * Tests the readColumns() method with an odd number of escape characters
+	 * char.
+	 */
+	@Test(expected = SuperCsvException.class)
+	public void testOddSeriesOfEscapeChars() throws Exception {
+
+		final CsvPreference csvPref = new CsvPreference.Builder(NORMAL_PREFERENCE)
+				.setQuoteEscapeChar('\\')
+				.build();
+
+		final String input = "\"\\\\\\\\\\\"";
+		tokenizer = createTokenizer(input, csvPref);
+		tokenizer.readColumns(columns);
+	}
+
+	/**
+	 * Tests the readColumns() method with a lone escape char outside of quoted fields
+	 */
+	@Test
+	public void testLoneEscapeCharInUnquotedField() throws Exception {
+		final CsvPreference csvPref = new CsvPreference.Builder(NORMAL_PREFERENCE)
+				.setQuoteEscapeChar('\\')
+				.build();
+
+		final String input = "one\\,\\two,thr\\ee";
+		tokenizer = createTokenizer(input, csvPref);
+		tokenizer.readColumns(columns);
+
+		assertTrue(columns.size() == 3);
+		assertEquals("one\\", columns.get(0));
+		assertEquals("\\two", columns.get(1));
+		assertEquals("thr\\ee", columns.get(2));
+		assertEquals(input, tokenizer.getUntokenizedRow());
+	}
+
+	/**
+	 * Tests the readColumns() method with a doubled escape char outside of quoted fields
+	 */
+	@Test
+	public void testDoubledEscapeCharInUnquotedField() throws Exception {
+		final CsvPreference csvPref = new CsvPreference.Builder(NORMAL_PREFERENCE)
+				.setQuoteEscapeChar('\\')
+				.build();
+
+		final String input = "one\\\\,\\\\two,thr\\\\ee";
+		tokenizer = createTokenizer(input, csvPref);
+		tokenizer.readColumns(columns);
+
+		assertTrue(columns.size() == 3);
+		assertEquals("one\\\\", columns.get(0));
+		assertEquals("\\\\two", columns.get(1));
+		assertEquals("thr\\\\ee", columns.get(2));
+		assertEquals(input, tokenizer.getUntokenizedRow());
+	}
+
+	/**
+	 * Tests the readColumns() method with an escape character preceding neither another escape
+	 * char nor a quote char.  In this situation, just pass the data through rather than
+	 * attempting to interpret the quote char.
+	 */
+	@Test
+	public void testEscapedNonEscapeChar() throws Exception {
```

Comment:
Non-multiline fields can be optionally wrapped in quotes: `col1,"col2"` would parse out to two fields: `col1` and `col2`.  The `testLoneEscapeCharInUnquotedField` test covers the first case (an unquoted field) and this `testEscapedNonEscapeChar` test covers the second case (a quoted field).

This maps to the state in `TokenizerState`: it's either `NORMAL` (not in a quoted field), or `QUOTE_MODE` (in a quoted field)